### PR TITLE
Rework no render and flags

### PR DESCRIPTION
**Pull request by mwestphal**
- Use f3d::window::Type instead of flags for engine
- add vtkF3DNoRenderWindow
- Simplify no-render implementation

### CODE REVIEW

File: library/src/window_impl.cxx [hunk 1]
```diff
@@ -1,49 +1,321 @@
...
+  if (type == WindowType::NO_RENDER)
+  {
+    this->Internals->RenWin = vtkSmartPointer<vtkF3DNoRenderWindow>::New();
+  }
+  else if (type == WindowType::EXTERNAL)
+  {
+    vtkNew<vtkExternalOpenGLRenderWindow> renWin;
+    renWin->AutomaticWindowPositionAndResizeOff();
+    this->Internals->RenWin = renWin;
+    this->Internals->RenWin->SetMultiSamples(0); // Disable hardware antialiasing
```
Comment (mwestphal) [inline library/src/window_impl.cxx:113]: is this needed @Meakk ?

File: .github/test_sdk/main.cxx [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 int main()
 {
-  f3d::engine eng(f3d::engine::FLAGS_NONE);
+  f3d::engine eng(f3d::window::WindowType::NO_RENDER);
```
Comment (Meakk) [inline .github/test_sdk/main.cxx:11]: `window::WindowType` is weird. What about renaming the enum just `type` or `Type`?

File: library/public/engine.h [hunk 3]
```diff
@@ -47,29 +47,13 @@ class F3D_EXPORT engine
...
-    CREATE_WINDOW = 1 << 0,     // 0001
-    CREATE_INTERACTOR = 1 << 1, // 0010
-    WINDOW_OFFSCREEN = 1 << 2,  // 0100
-    WINDOW_EXTERNAL = 1 << 3    // 1000
-  };
-
-  /**
-   * Engine constructor, configure it using the binary flags
+   * Engine constructor, choose the window type using the enum.
+   * see image.h for details about the window.
```
Comment (Meakk) [inline library/public/engine.h:51]: `image.h`?

File: library/public/engine.h [hunk 4]
```diff
@@ -47,29 +47,13 @@ class F3D_EXPORT engine
...
+   * When using ImageType::NO_RENDER, window and interactor will not be provided by the engine.
```
Comment (Meakk) [inline library/public/engine.h:52]: `ImageType`?

File: library/public/window.h [hunk 5]
```diff
@@ -19,6 +19,28 @@ namespace f3d
...
+   * =====================================
+   * NO_RENDER: A mock window without rendering capabilities
+   * NATIVE_WINDOW: A window using the native graphical stack
+   * NATIVE_OFFSCREEN: A native window rendering to an offscreen buffer, not visible on screen
+   * EXTERNAL_WINDOW: An external window that assume the OpenGL context would have been created by
+   * another framework
+   */
+  enum class WindowType : unsigned char
+  {
+    NO_RENDER,
```
Comment (Meakk) [inline library/public/window.h:33]: I would prefer `NONE`

File: library/src/window_impl.cxx (hunk 1 above)
Comment (Meakk) [reply library/src/window_impl.cxx:113]: Not for the external window, you can remove it

**Review by Meakk — CHANGES_REQUESTED** [review]

**Review by Meakk — APPROVED** [review]

**Pull request merged**